File: uoo6uoow/Week13/소수 구하기.java
import java.io.*;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        String[] input = reader.readLine().split(" ");
        int m = Integer.parseInt(input[0]);
        int n = Integer.parseInt(input[1]);

        boolean[] prime = eratosthenes(n);

        for (int i = m; i <= n; i++) {
            if (prime[i]){
                writer.write(String.valueOf(i + "\n"));
            }
        }
        writer.flush();
    }

    public static boolean[] eratosthenes(int max){
        boolean isPrime[] = new boolean[max + 1];
        Arrays.fill(isPrime, true); //소수라고 가정
        isPrime[0] = false;
        isPrime[1] = false; //소수 x

        for (int i = 2; i * i <= max; i++) {
            if (isPrime[i]){
                for (int j = i * i; j <= max ; j += i) {//i의 배수들을 제거
                    isPrime[j] = false;
                }
            }
        }

        return isPrime;
    }
}

File: noeyeyh/Week13/소수 구하기.py
import sys
n = 10 ** 6
isPrime = [True for x in range(n+1)]
isPrime[0], isPrime[1] = False, False

for i in range(2, int(n**0.5)+1):
    if isPrime[i]:
        for j in range(i*2, n+1, i):
            isPrime[j] = False

m, n = map(int, sys.stdin.readline().split())
for i in range(m, n+1):
    if isPrime[i]:
        print(i)